tests/tests.go

```diff
@@ -1316,11 +1330,17 @@ func (ctl *controller) waitBackup(databaseID, backupID int) error {
 		if backup == nil {
 			return errors.Errorf("backup %v for database %v not found", backupID, databaseID)
 		}
-		switch backup.Status {
-		case api.BackupStatusDone:
-			return nil
-		case api.BackupStatusFailed:
-			return errors.Errorf("backup %v for database %v failed", backupID, databaseID)
+		if cond != nil {
+			if cond(backup) {
+				return nil
+			}
+		} else {
```

Comment:
The `else` seems to be redundant.

Reply:
It's actually not.
If condition function is not nil and condition is not satisfied, we do not want to go back and check the backup.Status.
We want to use one of condition function/backup.Status for the check.